semantics/src/main/scala/aqua/semantics/rules/types/TypesAlgebra.scala

```diff
@@ -129,7 +129,7 @@ trait TypesAlgebra[S[_], Alg[_]] {
 
   def typeToCollectible(token: Token[S], givenType: Type): OptionT[Alg, CollectibleType]
 
-  def typeToStream(token: Token[S], givenType: Type): OptionT[Alg, StreamType]
+  def typeToStream(token: Token[S], givenType: Type, isMap: Boolean): OptionT[Alg, MutableStreamType]
```

Comment:
nit: Personally, I don't like such bool flags that essentially separate two distinct methods inside.
Here it is just `typeToStream(...): ...[..., StreamType]` and `typeToStreamMap(...): ...[..., StreamMapType]`.